### DISCUSSION TITLE
image to color correction node

### DISCUSSION THREAD

**Discussion by Ferniclestix** [WASasquatch/was-node-suite-comfyui#92] [Ideas]
would really love a node that uses another image for color correction.
This would be useful for multi sampler upscales due to samplers not respecting the source color profile. (this is solved in some other SD UI and is super useful for both inpainting and crop/paste situations.)

although it could probably be built into a sampler instead of being a custom node.

Example of what Im talking about  - the cropped and seperated image lower right has different colors on each after sampling :<
<img width="1652" alt="multiupscaler" src="https://github.com/WASasquatch/was-node-suite-comfyui/assets/128430314/b1dcfe92-706b-4044-9414-e651391b5be3">

**Comment by WASasquatch** [comment]
You could use Image Blending Mode and use color overlay like in Photoshop.

**Reply by Ferniclestix** [reply]
mmm ill look into it.  had a fair crack at it already and had no joy, always something wrong with it

**Reply by WASasquatch** [reply]
crack at what? Trying to create a node?

**Reply by Ferniclestix** [reply]
trying to get what already exists in comfyUI to do color correction. - the problem is, you can come close but it always seems to be a bespoke solution for a particular image and thats not ideal.

**Reply by WASasquatch** [reply]
> trying to get what already exists in comfyUI to do color correction. - the problem is, you can come close but it always seems to be a bespoke solution for a particular image and thats not ideal.

Let me try doing some research on the matter. Maybe their a raster solution for it with lookups.

**Reply by WASasquatch** [reply]
Really no option but color transfer, and that only assumes the images are very identical. Best I could do is a node that also considers the color differences, and only adjusts the colors if they're within a definable threshold. Could be useful.
